tools/migrate-lp-user-to-github

```diff
@@ -179,21 +179,27 @@ def main():
         subp(['git', 'clone',
               LP_UPSTREAM_PATH_TMPL.format(launchpad_user=args.launchpad_user),
               repo_dir])
+    cwd = os.getcwd()
     os.chdir(repo_dir)
     log("Sycing master branch with upstream")
     subp(['git', 'checkout', 'master'])
     subp(['git', 'pull'])
-    lp_remote_name, gh_remote_name = add_lp_and_github_remotes(
-        args.launchpad_user, args.github_user)
-    commit_msg = COMMIT_MSG_TMPL.format(
-        gh_username=args.github_user, lp_username=args.launchpad_user)
-    create_migration_branch(
-        MIGRATE_BRANCH_NAME, args.upstream, args.launchpad_user,
-        args.github_user, commit_msg)
-
-    for push_remote in (lp_remote_name, gh_remote_name):
-        subp(['git', 'push', push_remote, MIGRATE_BRANCH_NAME, '--force'])
-
+    try:
+        lp_remote_name, gh_remote_name = add_lp_and_github_remotes(
+            args.launchpad_user, args.github_user)
+        commit_msg = COMMIT_MSG_TMPL.format(
+            gh_username=args.github_user, lp_username=args.launchpad_user)
+        create_migration_branch(
+            MIGRATE_BRANCH_NAME, args.upstream, args.launchpad_user,
+            args.github_user, commit_msg)
+
+        for push_remote in (lp_remote_name, gh_remote_name):
+            subp(['git', 'push', push_remote, MIGRATE_BRANCH_NAME, '--force'])
```

Comment:
I would make the directory cleanup part of a `finally` clause further down, rather than handling both hear and on line 231.

Reply:
Oh, but wait! You allow the local developer to specify `--local-repo-dir`. Better make sure you only clean up the directory if you were the one who created it!